sktime/forecasting/base/_sktime.py

```diff
@@ -495,6 +495,64 @@ def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
         raise NotImplementedError("abstract method")
 
 
+class _OptionalDictParamsMixin:
```

Comment:
Do we still need this? If we want to keep it, this seems to be a major API change ...